src/widgets/block.rs

```diff
@@ -525,194 +525,238 @@ impl Widget for Block<'_> {
 
 impl WidgetRef for Block<'_> {
     fn render_ref(&self, area: Rect, buf: &mut Buffer) {
+        let area = area.intersection(buf.area);
```

Comment:
Just curious, why is this needed?

Reply:
Prevents a panics for blocks out of the buffer